pkg/transformers/json/transformer.go

```diff
@@ -14,21 +14,34 @@ import (
 
 const sep = "/"
 
-var keys = [...]string{"publisher", "protocol", "channel", "subtopic"}
-
 var (
+	keys = [...]string{"publisher", "protocol", "channel", "subtopic"}
+
 	// ErrTransform represents an error during parsing message.
-	ErrTransform         = errors.New("unable to parse JSON object")
-	ErrInvalidKey        = errors.New("invalid object key")
+	ErrTransform = errors.New("unable to parse JSON object")
+	// ErrInvalidKey represents the use of a reserved message field.
+	ErrInvalidKey = errors.New("invalid object key")
+
 	errUnknownFormat     = errors.New("unknown format of JSON message")
 	errInvalidFormat     = errors.New("invalid JSON object")
 	errInvalidNestedJSON = errors.New("invalid nested JSON object")
+
+	timestamps = []Timestamps{}
 )
 
 type funcTransformer func(messaging.Message) (interface{}, error)
 
+// Timestamps represents the message fields to use as timestamp
+type Timestamps struct {
```

Comment:
Don't use the plural (it's not a collection) and consider renaming to something like `TimeField`.